### DISCUSSION TITLE
Need help understanding the IPv6 addressing

### DISCUSSION THREAD

**Discussion by ValNyrov** [openthread/openthread#8240] [Q&A] [answered]
Hello OpenThread community!
I need help understanding addressing. I do codelab "Thread Border Router - Bidirectional IPv6 Connectivity and DNS-Based Service Discovery". (https://openthread.io/codelabs/openthread-border-router#0)
My hardware configuration is:
1)Working laptop 
2)Raspberry Pi 3 Model B + RCP (nrf52840 dongle), acting as border router
3)two boards nrf52840-DK acting as FTD
4)home Wi-Fi router

on page 3 of the codelab there is an  OMR address of the OTBR:

![codelab_0](https://user-images.githubusercontent.com/111975619/194133771-8697d4d9-1293-424d-a918-5eb5561249de.png)

on page 4 of the codelab there is an OMR address of the SRP client:

![codelab_1](https://user-images.githubusercontent.com/111975619/194134038-7accaa58-232f-4cb1-a1e0-ac66e50a3a6f.png)

The issue is that I don't see the OMR address on my FTD devices. My hardware configuration is:
1)Working laptop 
2)Raspberry Pi 3 Model B + RCP (nrf52840 dongle), acting as border router
3)two boards nrf52840-DK acting as FTD
4)home Wi-Fi router

Data from  Raspberry Pi:

**\> state
leader
Done**

**\> dataset active
Active Timestamp: 1
Channel: 14
Channel Mask: 0x07fff800
Ext PAN ID: 2ce31f93de11f7ca
Mesh Local Prefix: fdbd:f158:8d1f:6d0e::/64
Network Key: 80ed9837bd85bf55ac286a823fde257c
Network Name: my_ot_net
PAN ID: 0xabcd
PSKc: f3d17ce4893d79369f9b4a450bda353e
Security Policy: 672 onrc
Done**

**\> netdata show
Prefixes:
fdef:31c7:4351:1::/64 paos low 3400
Routes:
fd2c:e31f:93de:f7ca::/64 s med 3400
Services:
44970 01 5f000500000e10 s 3400
44970 5d fdbdf1588d1f6d0ed431095d2858bedad120 s 3400
Done**

**\> ipaddr
fdbd:f158:8d1f:6d0e:0:ff:fe00:fc11
fdef:31c7:4351:1:5f4:dfa8:ea4d:5e13**  \<- Did I understand correctly that is this the OMR address?
**fdbd:f158:8d1f:6d0e:0:ff:fe00:fc10
fdbd:f158:8d1f:6d0e:0:ff:fe00:fc38
fdbd:f158:8d1f:6d0e:0:ff:fe00:fc00
fdbd:f158:8d1f:6d0e:0:ff:fe00:3400
fdbd:f158:8d1f:6d0e:d431:95d:2858:beda
fe80:0:0:0:7868:cf70:4d1a:6cf6
Done**

**\> neighbor table**
**| Role | RLOC16 | Age | Avg RSSI | Last RSSI |R|D|N| Extended MAC     |**
**+------+-------------+-----+----------+-----------+-+-+-+------------------+**
**|   R   | 0x4800    |  10 |      -53 |       -51 |1|1|1| 2264b4f07fe29109 |**
**|   R   | 0xcc00    |   8  |      -49 |       -49 |1|1|1| 9ec7fd2017da05cb |**
**

Data from nrf52840-DK board 1:

**\> state
router
Done**

**\> dataset active
Active Timestamp: 1
Channel: 14
Channel Mask: 0x07fff800
Ext PAN ID: 2ce31f93de11f7ca
Mesh Local Prefix: fdbd:f158:8d1f:6d0e::/64
Network Key: 80ed9837bd85bf55ac286a823fde257c
Network Name: my_ot_net
PAN ID: 0xabcd
PSKc: f3d17ce4893d79369f9b4a450bda353e
Security Policy: 672 onrc
Done**

**\> netdata show
Prefixes:
fdef:31c7:4351:1::/64 paos low 3400
Routes:
fd2c:e31f:93de:f7ca::/64 s med 3400
Services:
44970 01 5f000500000e10 s 3400
44970 5d fdbdf1588d1f6d0ed431095d2858bedad120 s 3400
Done**

**\> ipaddr
fdbd:f158:8d1f:6d0e:0:ff:fe00:cc00
fdbd:f158:8d1f:6d0e:acec:29af:4975:d491
fe80:0:0:0:9cc7:fd20:17da:5cb
Done**

Here I don't see the OMR address, why?

**\> neighbor table**
**| Role | RLOC16 | Age | Avg RSSI | Last RSSI |R|D|N| Extended MAC     |**
**+------+--------+-----+----------+-----------+-+-+-+------------------+**
**|   R  | 0x3400 |  20 |      -46 |       -45 |1|1|1| 7a68cf704d1a6cf6 |**
**|   R  | 0x4800 |  20 |      -38 |       -37 |1|1|1| 2264b4f07fe29109 |**

Data from nrf52840-DK board 2:

**\> state
router
Done**

**\> dataset active
Active Timestamp: 1
Channel: 14
Channel Mask: 0x07fff800
Ext PAN ID: 2ce31f93de11f7ca
Mesh Local Prefix: fdbd:f158:8d1f:6d0e::/64
Network Key: 80ed9837bd85bf55ac286a823fde257c
Network Name: my_ot_net
PAN ID: 0xabcd
PSKc: f3d17ce4893d79369f9b4a450bda353e
Security Policy: 672 onrc
Done**

**\> netdata show
Prefixes:
fdef:31c7:4351:1::/64 paos low 3400
Routes:
fd2c:e31f:93de:f7ca::/64 s med 3400
Services:
44970 01 5f000500000e10 s 3400
44970 5d fdbdf1588d1f6d0ed431095d2858bedad120 s 3400
Done**

**\> ipaddr
fdbd:f158:8d1f:6d0e:0:ff:fe00:4800
fdbd:f158:8d1f:6d0e:ee5d:2a8c:c75a:4079
fe80:0:0:0:2064:b4f0:7fe2:9109
Done**

Here I don't see the OMR address, why?

**\> neighbor table**
**| Role | RLOC16 | Age | Avg RSSI | Last RSSI |R|D|N| Extended MAC     |**
**+------+--------+-----+----------+-----------+-+-+-+------------------+**
**|   R  | 0x3400 |   9 |      -51 |       -50 |1|1|1| 7a68cf704d1a6cf6 |**
**|   R  | 0xcc00 |  21 |      -37 |       -37 |1|1|1| 9ec7fd2017da05cb |**

And I don't understand yet how I can ping my FTD devices( nrf52840-DK boards) from a laptop?

Best Regards,
Valeriy

**Comment by jwhui** [comment, marked as answer]
It appears that [`OPENTHREAD_CONFIG_IP6_SLAAC_ENABLE`](https://github.com/openthread/openthread/blob/c5b4cd7bbc14912af2737ace7a7e4052aa76690c/src/core/config/ip6.h#L117) is not enabled. Which would explain why your nRF52840-DK devices are not generating SLAAC addresses out of the prefixes contained in the Thread Network Data.

Can you provide more detail on your nRF52840-DK firmware? How did you build the nRF52840-DK firmware? Did you follow the steps in [Set Up SRP Client End Device](https://openthread.io/codelabs/openthread-border-router#3) section of the codelab?

**Reply by ValNyrov** [reply]
Hi Jonathan!
Thank you for your support!  I use ZephyrOS on the nRF52840-DK board and built the firmware myself, perhaps unknowingly, I made a mistake somewhere in the configuration of the OpenThread module.
I got to step [4. Set Up SRP Client End Device](https://openthread.io/codelabs/openthread-border-router#3), but I was confused by the lack of an OMR address. I'll try to figure it out.

**Reply by ValNyrov** [reply]
> It appears that [`OPENTHREAD_CONFIG_IP6_SLAAC_ENABLE`]

You are absolutely right, I didn't include the CONFIG_OPENTHREAD_SLAAC=y option. Thank You!

**Reply by ValNyrov** [reply]
Dear Jonathan!
I successfully completed the laboratory work. I can ping and send UDP packets to Thread devices inside my home network.
(also successfully registered services).  I don't yet understand how to send and receive data from Thread devices outside my home network, and I need a hint-which technology to use for this purpose. Maybe it's CoAP, MQTT or there's something else? Please advise me what I need to study. Thank you.

Best Regards,
Valeriy
